Tests/test_Restriction.py

```diff
@@ -303,8 +519,75 @@ def test_analysis_restrictions(self):
         # Fix negative start:
         self.assertEqual((ana._boundaries(-1, 20)[:2]), (20, 33))
+        # Fix negative end:
+        self.assertEqual((ana._boundaries(1, -1)[:2]), (1, 33))
+        # Sites in- and outside of boundaries
+        new_seq = Seq('GAATTCAAAAAAGAATTC', IUPACAmbiguousDNA())
+        rb = RestrictionBatch([EcoRI])
+        ana = Analysis(rb, new_seq)
+        # Cut at least inside
+        self.assertEqual(ana.between(1, 7), {EcoRI: [2, 14]})
+        # Cut at least inside and report only inside site
+        self.assertEqual(ana.show_only_between(1, 7), {EcoRI: [2]})
+        # Cut at least outside
+        self.assertEqual(ana.outside(1, 7), {EcoRI: [2, 14]})
+        # Don't cut within
+        self.assertEqual(ana.do_not_cut(7, 12), {EcoRI: [2, 14]})
+
+
+class TestPrintOutputs(unittest.TestCase):
+    """Class to test various print outputs."""
+
+    import sys
+
+    try:  # Python 2
+        from StringIO import StringIO
+    except ImportError:  # Python 3
+        from io import StringIO
+
+    def test_supplier(self):
+        """Test output of supplier list for different enzyme types."""
+        out = self.StringIO()
+        self.sys.stdout = out
+        EcoRI.suppliers()
+        self.assertIn('Life Technologies', out.getvalue())
+        self.assertEqual(SnaI.suppliers(), None)
+        EcoRI.all_suppliers()  # Independent of enzyme, list of all suppliers
+        self.assertIn('Agilent Technologies', out.getvalue())
+        batch = EcoRI + SnaI
+        batch.show_codes()
+        self.assertIn('N = New England Biolabs', out.getvalue())
+        self.sys.stdout = self.sys.__stdout__
+
+    def test_print_that(self):
+        """Test print_that function."""
+        out = self.StringIO()
+        self.sys.stdout = out
+        my_batch = EcoRI + SmaI + KpnI
+        my_seq = Seq('GAATTCCCGGGATATA')  # EcoRI and SmaI sites
+        analysis = Analysis(my_batch, my_seq)
+        analysis.print_that(None, title='My sequence\n\n',
+                            s1='Non Cutters\n\n')
+        self.assertIn('My sequence', out.getvalue())
+        self.assertIn('Non Cutters', out.getvalue())
+        self.assertIn('2.', out.getvalue())
+        self.sys.stdout = self.sys.__stdout__
+
+    def test_str_method(self):
+        """Test __str__ and __repr__ outputs."""
+        batch = EcoRI + SmaI + KpnI
+        self.assertEqual(str(batch), 'EcoRI+KpnI+SmaI')
+        batch += Asp718I
+        batch += SnaI
+        self.assertEqual(str(batch), 'Asp718I+EcoRI...SmaI+SnaI')
+        self.assertEqual(repr(batch),
+                         "RestrictionBatch(['Asp718I', 'EcoRI', 'KpnI', "
+                         "'SmaI', 'SnaI'])")
 
 
 if __name__ == "__main__":
     runner = unittest.TextTestRunner(verbosity=2)
     unittest.main(testRunner=runner)
+    batch = EcoRI + SmaI + KpnI
+    my_seq = Seq('GAATTCCCGGGATGCGTA')
+    analysis = Analysis(batch, my_seq)
```

Comment:
Can you remove this little self test? It would not be used via ``run_tests.py``?